src/backend/vm/runtime.rs

```diff
@@ -147,124 +272,58 @@ impl VM {
-                        (Float(i1), Float(i2)) => Float(i1 / i2),
-                        (type1, type2) => {
-                            return Err(self
-                                .on_error(format!(
-                                    "Division can not be performed between {} and {}",
-                                    get_type_name(&type1),
-                                    get_type_name(&type2)
-                                ))
-                                .unwrap_err());
-                        }
-                    };
-                    self.save_register(rd, rd_data)?;
+                    self.exec_arithmetic_float("Division", rs1, rs2, rd, f64::div, Float)?;
                 }
                 idiv => {
-                    let rs1_data = self.load_register(rs1)?;
-                    let rs2_data = self.load_register(rs2)?;
-                    let rd_data = match (rs1_data, rs2_data) {
-                        (Int(i1), Int(i2)) => i1 as f64 / i2 as f64,
-                        (Float(i1), Int(i2)) => i1 / i2 as f64,
-                        (Int(i1), Float(i2)) => i1 as f64 / i2,
-                        (Float(i1), Float(i2)) => i1 / i2,
-                        (type1, type2) => {
-                            return Err(self
-                                .on_error(format!(
-                                    "Floor division can not be performed between {} and {}",
-                                    get_type_name(&type1),
-                                    get_type_name(&type2)
-                                ))
-                                .unwrap_err());
-                        }
-                    };
-                    let rd_data = Data::Int(rd_data as i64);
-                    self.save_register(rd, rd_data)?;
+                    let gen_data = |x: f64| Int(x.floor() as i64);
+                    self.exec_arithmetic_float("Floor division", rs1, rs2, rd, f64::div, gen_data)?;
+                }
+                modu => {
+                    use std::ops::Rem;
+                    self.exec_arithmetic_int_float("Modulus", rs1, rs2, rd, i64::checked_rem, f64::rem)?;
+                }
+                exp => {
+                    self.exec_arithmetic_float("Exponential", rs1, rs2, rd, f64::powf, Float)?;
+                }
+                or => {
+                    use std::ops::BitOr;
+                    self.exec_arithmetic_two_bool("Logical or", rs1, rs2, rd, bool::bitor)?;
+                }
+                and => {
+                    use std::ops::BitAnd;
+                    self.exec_arithmetic_two_bool("Logical and", rs1, rs2, rd, bool::bitand)?;
+                }
+                not => {
+                    use std::ops::Not;
+                    self.exec_arithmetic_one_bool("Logical Not", rs1, rd, bool::not)?;
+                }
+                ge => {
+                    self.exec_arithmetic_compare("GE", rs1, rs2, rd, i64::ge, f64::ge)?;
```

Comment:
`GE` is short for `greater or equal to`. We may want to change the naming here to `">="` or `Greater than or equal to`.
```suggestion
                    self.exec_arithmetic_compare("\">=\"", rs1, rs2, rd, i64::ge, f64::ge)?;
```

Reply:
Edited. Please see [give symbols to compare operators in error message](https://github.com/diatom-lang/diatom/pull/3/commits/e976ea624534bc0a61b121ef24b0fb88f84f9abe).